docs/data/data-grid/components/toolbar/toolbar.md

```diff
@@ -0,0 +1,91 @@
+---
+title: React Data Grid - Toolbar component
+productId: x-data-grid
+components: ToolbarRoot, ToolbarButton
+packageName: '@mui/x-data-grid'
+githubLabel: 'component: data grid'
+---
+
+# Data Grid - Toolbar
+
+<p class="description">Add custom actions and controls to the Data Grid.</p>
+
+## Basic usage
+
+The demo below shows the default toolbar configuration.
+
+To extend the default toolbar, you can use copy the code below and customize it to your needs.
+
+{{"demo": "GridToolbar.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Anatomy
+
+```tsx
+import { Toolbar } from '@mui/x-data-grid';
+
+<Toolbar.Root>
+  <Toolbar.Button />
+</Toolbar.Root>;
+```
+
+### Root
+
+The top level component that provides context to child components.
+
+Renders a styled `<div />` element.
+
+### Button
+
+A button that can be used to perform actions from the toolbar.
+
+Renders the `baseIconButton` slot.
+
+## Recipes
+
+Below are some ways the Toolbar component can be used.
+
+### Settings menu
+
+Allow users to customize the data grid appearance from the toolbar.
+
+{{"demo": "GridToolbarSettingsMenu.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Filter bar
+
+Show active filter chips in the toolbar.
+
+{{"demo": "GridToolbarFilterBar.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Custom elements
+
+The default elements can be replaced using the `render` prop. See [Grid components—Customization](/x/react-data-grid/components/overview/#customization).
+
+The demo below shows how to replace the default elements with custom ones, styled with Tailwind CSS.
+
+{{"demo": "GridToolbarCustom.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
```suggestion
{{"demo": "GridToolbarCustom.js", "bg": "inline", "defaultCodeOpen": false, "iframe": true}}
```

Styles break in an iframe currently, When CSS vars revert PR is merged this need to change https://github.com/mui/mui-x/pull/16173